src/components/catalog/list.js

```diff
@@ -62,61 +63,70 @@ const ArrowUp = styled.div`
   -webkit-transform: rotate(-135deg);
 `;
 
-class List extends React.Component {
-  constructor(props) {
-    super(props);
-    this.state = {
-      isOpened: false,
-    };
-    this.toggle = this.toggle.bind(this);
-    this.renderItems = this.renderItems.bind(this);
-  }
-
-  toggle() {
-    this.setState({ isOpened: !this.state.isOpened });
-  }
-
-  renderItems(items) {
-    const { padding, onSelect } = this.props;
-
-    return items.map((item, i) => {
-      if (item.children) {
-        return <List key={i} name={item.name} items={item.children} onSelect={onSelect} padding={padding + 16} />;
+export default class List extends React.Component {
+  static propTypes = {
+    name: PropTypes.string.isRequired,
+    methods: PropTypes.array.isRequired,
+    padding: PropTypes.number.isRequired,
+    selectMethod: PropTypes.func.isRequired,
+    selectedMethod: PropTypes.object,
+  };
+
+  static defaultProps = {
+    padding: 10,
+  };
+
+  state = {
+    selectedMethod: this.props.selectedMethod,
```

Comment:
why do you need to have two sources of **selectedMethod** - state and props? why not only props?

Reply:
because I want to persist this selected method selection another method -> should be applied styles for another method.